ERCS/erc-7855.md

```diff
@@ -0,0 +1,149 @@
+---
+eip: 7855
+title: Fungible Token with gas rebates
+description: Built-in gas rebates, secure signature validation, and optimized transfer mechanisms.
+author: Amitesh Gautam (@blockchainDevAmitesh) <blockchaindev.amitesh@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7849-fungible-token-with-gas-rebates/22356
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-01-01
+---
+
+## Abstract
+
+This proposal introduces a new fungible token standard focused on security, efficiency, and interoperability. It extends traditional [ERC-20](./eip-20.md) functionality with built-in gas rebates, secure signature validation using [EIP-712](./eip-712.md), and optimized transfer mechanisms. The standard includes protections against common attack vectors while maintaining full compatibility with existing token interfaces.
+
+## Motivation
+
+Current fungible token standards lack built-in security features and gas optimization mechanisms. This proposal addresses these limitations by incorporating:
+- Native gas rebate system to reduce transaction costs
+- [EIP-712](./eip-712.md) compliant signatures for gasless approvals
+- Strong security measures against common vulnerabilities
+- Efficient state management and error handling
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Core Variables
+- `name`: Token name
+- `symbol`: Token symbol
+- `decimals`: Number of decimal places
+- `totalSupply`: Total token supply
+- `_balances`: Mapping of address to token balance
+- `_allowances`: Mapping of owner address to spender address to allowed amount
```

Comment:
"Variables" aren't really a thing in contract ABIs. Everything is a function. You should only be defining the externally visible behaviour of your standard, not implementation details. Whether these are tracked as state slots, hard coded, or whatever else should not matter to your standard.